tools/pipelines/test-perf-benchmarks-using-built-artifacts.yml

```diff
@@ -0,0 +1,376 @@
+          inputs:
+            workingFile: ${{ variables.testWorkspace }}/.npmrc
+            customEndpoint: 'Office Packages - Readonly'
+
+        - task: Bash@3
+          displayName: Print Parameters
+          inputs:
+            targetType: 'inline'
+            script: |
+              echo "
+              Pipeline Parameters:
+                poolBuild=${{ parameters.poolBuild }}
+                Build=${{ parameters.taskBuild }}
+
+              Task Variables:
+                releaseBuild=$(releaseBuildVar)
+                toolAbsolutePath=$(toolAbsolutePath)
+                artifactPipeline=$(artifactPipeline)
+                feed=$(feed)
+                BuildDir=${{ variables.buildDirectory }}
+                TestWorkspace=${{ variables.testWorkspace }}
+
+              Build Params
+                SourceBranch=$(Build.SourceBranch)
+              "
+
+        - task: Bash@3
+          displayName: 'Install aria-logger'
+          inputs:
+            targetType: 'inline'
+            workingDirectory: $(toolAbsolutePath)
+            script: |
+              cp ${{ variables.testWorkspace }}/.npmrc . ;
+              npm install @ff-internal/aria-logger;
+              npm i;
+              npm run build:compile;
+
+        # Download Test Files
+        # Download Artifact - Test Files
+        - task: DownloadPipelineArtifact@2
+          displayName: Download test files
+          inputs:
+            source: specific
+            project: internal
+            pipeline: ${{ variables.artifactPipeline }}
+            preferTriggeringPipeline: true
+            allowPartiallySucceededBuilds: true
+            runVersion: latestFromBranch
+            runBranch: $(Build.SourceBranch)
+            artifact: test-files
+            path: $(Pipeline.Workspace)/test-files/
+
+        - ${{ each execTestPackage in parameters.executionTestPackages }}:
+
+          # Download package artifacts
+          - task: DownloadPipelineArtifact@2
+            displayName: Download test package
+            inputs:
+              source: specific
+              project: internal
+              pipeline: ${{ variables.artifactPipeline }}
+              preferTriggeringPipeline: true
+              allowPartiallySucceededBuilds: true
+              runVersion: latestFromBranch
+              runBranch: $(Build.SourceBranch)
+              artifact: pack
+              patterns: "**/${{ replace(replace(execTestPackage, '@', '' ), '/', '-') }}-*.tgz"
+              path: $(Pipeline.Workspace)/client/pack
+
+          # Install test package
+          - task: Bash@3
+            displayName: Install Test Package
+            inputs:
+              targetType: 'inline'
+              workingDirectory: ${{ variables.testWorkspace }}
+              script: |
+                echo "Install ${{ execTestPackage }}"
+                TEST_PACKAGE_FILE_PATTERN=${{ replace(replace(execTestPackage, '@', '' ), '/', '-') }}-*.tgz
```

Comment:
This seems too be repeated, should it be a pipeline variable?

Reply:
This is inside a for loop and we cannot re-use a variable name in the pipeline. I already went down that path sadly.

Reply:
meh... that's ok to keep it as is.